### ISSUE TITLE
SSE Support

### ISSUE THREAD

**Issue by JohannesHepp** [SMILEY4/ktor-openapi-tools#161]
Hey, 

ktor 3.0.0 introduces support for [SSE](https://api.ktor.io/ktor-client/ktor-client-core/io.ktor.client.plugins.sse/sse.html?_gl=1*8ba0be*_gcl_au*ODY5NTA3MDY2LjE3MzE5NDE0NjA.*_ga*ODM4Nzc1Nzc3LjE3MjA2Nzc5OTI.*_ga_9J976DJZ68*MTczMjEwNTk5OS4xNS4xLjE3MzIxMDYwMjQuMzUuMC4w). We already use this feature. We also want to automatically add these endpoints to our open API spec. It would be great if you could add support for SSE to the open API spec for your library :)

**Comment by SMILEY4** [comment]
Hi, sorry for the late response.

I'm not sure if openapi is a good fit for documenting this type of communication. Do you have an idea or suugestion on how this could look like in swagger?

**Comment by king-phyte** [comment]
@JohannesHepp If it is merely about discovery, I think the package already supports that. Here is an example and the equivalent Swagger UI displayed

```kotlin
routing {
        sse("/server-sent-events/som") {
            repeat(1000) {
                send(ServerSentEvent("e203-2032"))
                delay(1000)
            }
        }
    }
```

<img width="1261" alt="image" src="https://github.com/user-attachments/assets/d496b248-a456-4ea3-9f13-32726160cd11" />



@SMILEY4 I think we can add an extension function like we do got `get`, `post`, etc., for `sse` that allows one to descibe the requests and responses. That would help document it manually, like define security schemes and all other OpenAPI definitions we already have. For example, look at the code below and the equivalent Swagger UI, which is perfectly valid in Ktor:

```kotlin
routing {
        authenticate("user-jwt") {
            sse("/server-sent-events") {
                repeat(1000) {
                    val principal = call.principal<String>()
                    send(ServerSentEvent("Principal $principal"))
                    delay(1000)
                }
            }
        }
```

There is no way to add a security scheme to the above, so in Swagger, it appears like 
<img width="1261" alt="image" src="https://github.com/user-attachments/assets/1a440b78-2a51-42f0-a7bd-7d3cf28ddbb8" />

Notice there is no padlock icon beside the route.

If we modified it to become like

```kotlin
routing {
        authenticate("user-jwt") {
            sse("/server-sent-events", {
            securitySchemeNames("User")
        }) {
                repeat(1000) {
                    val principal = call.principal<String>()
                    send(ServerSentEvent("Principal $principal"))
                    delay(1000)
                }
            }
        }
```
one would be able to get something like

<img width="1247" alt="image" src="https://github.com/user-attachments/assets/a1b44c2f-2bce-45e2-ad23-1782f22ea830" />

Now notice the padlock icon, response definition and all

**Comment by JohannesHepp** [comment]
Yes the target would be to add some basic documentation like the correct tag, securitySchema,...

**Comment by SMILEY4** [comment]
ok yeah, i think that would be the way to go too or at least be a good first step and wouldn't require many changes

**Comment by SMILEY4** [comment]
Hi @JohannesHepp,

Sorry, I think i completely forgot about something :) Adding documentation to sse routes should already be completely supported. You can wrap any ktor-route in a `documentation` route and add your openapi documentation there

Example:
```kotlin
// ktor-swagger-ui documentation route
documentation({
    // all openapi documentation available here
    tags("sse")
    description = "Receive events from this endpoint"
    protected = true
    securitySchemeNames("basic-auth")
    response {
        "event" to {
            body<String> {
                mediaTypes(ContentType.Text.Plain)
            }
        }
    }
}) {
    // normal ktor-sse route
    sse("/events") {
        repeat(6) {
            send(ServerSentEvent("this is SSE #$it"))
            delay(1000)
        }
    }
}
```

![Image](https://github.com/user-attachments/assets/67651388-12da-4931-a80c-f531f99bf5d1)

**Comment by king-phyte** [comment]
@SMILEY4 That is awesome, but also completely new to me. Did I miss that from the docs or it wasn't mentioned in there?

Perhaps we can add a section for that in the new docs?
